### PR TITLE
[Keyboard] Added Atreyu keyboard.

### PR DESCRIPTION
**Pull request by climent**
Pull request for a new Atreyu keyboard to mainline.

## Description

Keyboard and keymap definition for a new atreyu keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/atreyu/rev1/rev1.c [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "atreyu.h"
```
Comment (drashna) [inline keyboards/atreyu/rev1/rev1.c:17]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) {
        return false; 
    }

    if (index == 1) {
      if (clockwise) {
          tap_code(KC_VOLU);
      } else {
          tap_code(KC_VOLD);
      }
    }
    if (index == 0) {
      if (IS_LAYER_ON(_LOWER)) {
        if (clockwise) {
            tap_code(KC_WH_U);
        } else {
            tap_code(KC_WH_D);
        }
      } else {
        if (clockwise) {
            tap_code(KC_VOLU);
        } else {
            tap_code(KC_VOLD);
        }
      }
    }
    return true;
}
#endif
```

File: keyboards/atreyu/rules.mk [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# MCU name
+MCU = atmega32u4
+
+DEFAULT_FOLDER = atreyu/rev1
+SPLIT_KEYBOARD = no
+ENCODER_ENABLE = yes
```
Comment (drashna) [inline keyboards/atreyu/rules.mk:6]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = atmel-dfu

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output

ENCODER_ENABLE = yes
```

File: keyboards/atreyu/readme.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# atreyu
+
+![atreyu](https://github.com/climent/atreyu/raw/main/pictures/PXL_20220331_122640681.jpg?raw=true)
```
[the remark this answers is not in the document]
Comment (climent) [reply keyboards/atreyu/readme.md]: Updated to imgur image.

File: keyboards/atreyu/readme.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+*An unsplit, modified baby of a Lily58 having a baby with a Sofle keyboard*
+
+* Keyboard Maintainer: [Jesus Climent](https://github.com/Jesus Climent)
+* Hardware Supported: AtreyuKeyboard PCB, ProMicro
+* Hardware Availability: [PCB and case data](https://github.com/climent/atreyu)
+
+Make example for this keyboard (after setting up your build environment):
```
[the remark this answers is not in the document]
Comment (climent) [reply keyboards/atreyu/readme.md:11]: Updated readme.md with suggestion.

**Comment by climent** [comment]
Please, take another look.

File: keyboards/atreyu/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+#define KC_FN FN
+#define KC_LWR LWR
+#define KC_RSE RSE
+#define KC_RST RESET
+#define KC_BL_S BL_STEP
+#define KC_BL_T BL_TOGG
+#define KC_LCTL_T LCTL_T
+#define KC_GUI_T GUI_T
+#define KC_ALT_T ALT_T
```
Comment (drashna) [inline keyboards/atreyu/keymaps/default/keymap.c:44]:
```suggestion
```

File: keyboards/atreyu/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+        return false;
+      }
+      return true;
+      break;
+  }
+  return true;
+}
+
+#ifdef ENCODER_ENABLE
+
```
Comment (drashna) [inline keyboards/atreyu/keymaps/default/keymap.c:123]:
```suggestion
```

File: keyboards/atreyu/readme.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
+# atreyu
+
+![atreyu](https://i.imgur.com/NP0cux1.jpg)
```
Comment (drashna) [inline keyboards/atreyu/readme.md:3]:
```suggestion
![atreyu](https://i.imgur.com/NP0cux1h.jpg)
```

**Review by drashna — CHANGES_REQUESTED** [review]
Is there a reason for the revision folder? Do you plan on adding additional versions of the board?

**Comment by climent** [comment]
> Is there a reason for the revision folder? Do you plan on adding additional versions of the board?

Yes, I am just completing rev2 which comes with an OLED screen, and requires pin changes.

**Review by drashna — APPROVED** [review]

File: keyboards/atreyu/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+  _______, KC_CAPS, _______, _______, _______, _______,                   _______, _______, _______, _______, _______, _______, \
+  _______,          _______, _______, _______, KC_SPC , KC_VOLD, KC_VOLU, _______, _______, _______, _______, _______\
+),
+
+[_ADJUST] = LAYOUT(\
+  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, \
+  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, \
+  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, \
+  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, \
+  _______,          _______, _______, _______, _______ , _______, _______ , _______, _______, _______, _______,         _______\
```
Comment (fauxpark) [inline keyboards/atreyu/keymaps/default/keymap.c:67]:
```suggestion
[_RAISE] = LAYOUT(
  _______, _______, _______, _______, _______, _______,                    KC_HOME, KC_END , _______, KC_LCBR, KC_RCBR, KC_TILD,
  _______, _______, _______, _______, KC_HOME, KC_END ,                    KC_PGUP, XXXXXXX, KC_UP  , KC_LBRC, KC_RBRC, KC_DEL ,
  _______, _______, _______, _______, _______, _______,                    KC_PGDN, KC_LEFT, KC_DOWN, KC_RGHT, KC_PIPE, KC_ENT ,
  _______, _______, _______, _______, _______, _______,                    KC_UNDS, KC_MINS, KC_EQL , KC_PLUS, KC_BSLS, _______,
  _______,          _______, _______, _______, KC_SPC , KC_HOME, KC_END  , _______, _______, KC_RGUI, KC_RALT,          _______
),

[_LOWER] = LAYOUT(
  KC_GRV , KC_F1  , KC_F2  , KC_F3  , KC_F4  , KC_F5  ,                   KC_F6  , KC_F7  , KC_F8  , KC_F9  , KC_F10 , _______,
  _______, KC_VOLU, _______, _______, TG(_RAISE), AG_TOGG,                _______, _______, _______, _______, _______, KC_DEL ,
  _______, KC_VOLD, _______, _______, _______, _______,                   _______, _______, _______, _______, _______, _______, 
  _______, KC_CAPS, _______, _______, _______, _______,                   _______, _______, _______, _______, _______, _______,
  _______,          _______, _______, _______, KC_SPC , KC_VOLD, KC_VOLU, _______, _______, _______, _______, _______
),

[_ADJUST] = LAYOUT(
  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
  _______,          _______, _______, _______, _______ , _______, _______ , _______, _______, _______, _______,         _______
```

File: keyboards/atreyu/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+[_QWERTY] = LAYOUT(
+     KC_ESC  , KC_1   , KC_2   , KC_3   , KC_4   , KC_5   ,                   KC_6   , KC_7   , KC_8   , KC_9   , KC_0    ,KC_GRV  ,
+     KC_TAB  , KC_Q   , KC_W   , KC_E   , KC_R   , KC_T   ,                   KC_Y   , KC_U   , KC_I   , KC_O   , KC_P    ,KC_BSPC ,
+     KC_LCTRL, KC_A   , KC_S   , KC_D   , KC_F   , KC_G   ,                   KC_H   , KC_J   , KC_K   , KC_L   , KC_SCLN ,KC_QUOT ,
+     KC_LSFT , KC_Z   , KC_X   , KC_C   , KC_V   , KC_B   ,                   KC_N   , KC_M   , KC_COMM, KC_DOT , KC_SLSH ,KC_RSFT ,
+     KC_LCTRL,          LWR    , KC_LALT, KC_LGUI, KC_ENT , KC_LBRC, KC_RBRC, KC_SPC , RSE    , KC_NO  ,  KC_NO ,          KC_RCTRL
```
Comment (fauxpark) [inline keyboards/atreyu/keymaps/default/keymap.c:42]:
```suggestion
     KC_LCTL,           LWR    , KC_LALT, KC_LGUI, KC_ENT , KC_LBRC, KC_RBRC, KC_SPC , RSE    , KC_NO  ,  KC_NO ,          KC_RCTL 
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/atreyu/rev1/info.json [hunk 10]
```diff
@@ -0,0 +1,76 @@
+{
+    "keyboard_name": "Atreyu",
+    "manufacturer": "Heyzeus",
+    "url": "https://github.com/climent/atreyu",
+    "maintainer": "Jesus Climent <climent@gmail.com>",
+    "usb": {
+      "vid": "0xFEED",
+      "pid": "0x0001",
+      "device_version": "0.0.1",
```
Comment (fauxpark) [inline keyboards/atreyu/rev1/info.json:9]:
```suggestion
      "device_version": "0.0.1"
```

File: keyboards/atreyu/readme.md [hunk 11]
```diff
@@ -0,0 +1,29 @@
+# atreyu
+
+![atreyu](https://i.imgur.com/NP0cux1h.jpg)
+
+*An unsplit, modified version of a Lily58 having a baby with a Sofle keyboard*
```
Comment (fauxpark) [inline keyboards/atreyu/readme.md:5]:
```suggestion
An unsplit, modified version of a Lily58 having a baby with a Sofle keyboard
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by climent** [comment]
Anything else needed to get this approved?

**Review by climent — COMMENTED** [review]
All changes merged.

**Review by keyboard-magpie — APPROVED** [review]
The picture including an encoder may be confusing for some, a future PR with an encoder enabled keymap may be a good idea.

**Pull request merged**